### PR TITLE
MGMT-9683: don't mark vsphere installation with Cluster-managed-netwo…

### PR DESCRIPTION
**Pull request by michaellevy101**
…rking-with-VMs

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -2625,26 +2625,51 @@ func (b *bareMetalInventory) setOperatorsUsage(updateOLMOperators []*models.Moni
 }
 
 func (b *bareMetalInventory) updateClusterNetworkVMUsage(cluster *common.Cluster, updateParams *models.V2ClusterUpdateParams, usages map[string]models.Usage, log logrus.FieldLogger) {
-	vmHosts := make([]string, 0)
+	platform := cluster.Platform
+	usageEnable := true
```
Comment (gamli75) [inline internal/bminventory/inventory.go:2629]: what is the meaning of this parameter? please add a comment

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2625,26 +2625,51 @@ func (b *bareMetalInventory) setOperatorsUsage(updateOLMOperators []*models.Moni
...
+	}
+
 	userManagedNetwork := cluster.UserManagedNetworking != nil && *cluster.UserManagedNetworking
+
 	if updateParams != nil && updateParams.UserManagedNetworking != nil {
 		userManagedNetwork = *updateParams.UserManagedNetworking
 	}
-	if !userManagedNetwork {
+
+	if userManagedNetwork {
```
Comment (gamli75) [inline internal/bminventory/inventory.go:2645]: please put all the code that changes the usageEnable to false together

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -2625,26 +2625,51 @@ func (b *bareMetalInventory) setOperatorsUsage(updateOLMOperators []*models.Moni
...
 func (b *bareMetalInventory) updateClusterNetworkVMUsage(cluster *common.Cluster, updateParams *models.V2ClusterUpdateParams, usages map[string]models.Usage, log logrus.FieldLogger) {
-	vmHosts := make([]string, 0)
+	platform := cluster.Platform
+	usageEnable := true
+
+	if updateParams != nil && updateParams.Platform != nil {
+		platform = updateParams.Platform
+	}
+
+	if *platform.Type != models.PlatformTypeBaremetal && *platform.Type != models.PlatformTypeNone {
```
Comment (gamli75) [inline internal/bminventory/inventory.go:2635]: can it be nil?

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (michaellevy101) [reply internal/bminventory/inventory.go:2629]: what parameter? The platform could be from the persisted cluster or just changed by the UI. all the code has this fallback.

usageEnable: means enable(create) or disable(delete) this usage.

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (gamli75) [reply internal/bminventory/inventory.go:2645]: at least the first 2 conditions :)

Comment (gamli75) [reply internal/bminventory/inventory.go:2645]: never mind - let's keep it as it is

**Review by gamli75 — APPROVED** [review]

**Pull request merged**